File: Models/ScanHistory.swift
import Foundation

struct ScanHistory: Codable, Hashable, Identifiable {
    var id: UUID
    var startedAt: Date
    var finishedAt: Date
    var ipRange: String
    var totalHosts: Int
    var foundDevices: Int
    var devices: [Device]

    init(
        id: UUID = UUID(),
        startedAt: Date,
        finishedAt: Date,
        ipRange: String,
        totalHosts: Int,
        foundDevices: Int,
        devices: [Device]
    ) {
        self.id = id
        self.startedAt = startedAt
        self.finishedAt = finishedAt
        self.ipRange = ipRange
        self.totalHosts = totalHosts
        self.foundDevices = foundDevices
        self.devices = devices
    }

    var duration: TimeInterval {
        finishedAt.timeIntervalSince(startedAt)
    }
}

File: Windows/LanScope.Core/Models/ScanHistory.cs
namespace LanScope.Core.Models;

public sealed class ScanHistory
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public DateTimeOffset StartedAt { get; set; }
    public DateTimeOffset FinishedAt { get; set; }
    public string IpRange { get; set; } = "";
    public int TotalHosts { get; set; }
    public int FoundDevices { get; set; }
    public List<Device> Devices { get; set; } = [];
    public double DurationSeconds => Math.Max(0, (FinishedAt - StartedAt).TotalSeconds);
    public string Summary => $"{StartedAt:g}  ·  {FoundDevices} found";
}
